TUM Campus App/CafeteriaMenuManager.swift

```diff
@@ -44,22 +48,91 @@ class CafeteriaMenuManager: Manager {
     }
     
     func addMenu(_ item: JSON) {
-        if let cafeteria = item["mensa_id"].string, let date = item["date"].string, let typeShort = item["type_short"].string, let typeLong = item["type_long"].string, let name = item["name"].string, let mensa = self.manager?.getCafeteriaForID(cafeteria) {
+        if let cafeteria = item["mensa_id"].string, let date = item["date"].string, let typeShort = item["type_short"].string, let typeLong = item["type_long"].string, var name = item["name"].string, let mensa = self.manager?.getCafeteriaForID(cafeteria) {
             let id = item["id"].string ?? ""
             let typeNR = item["type_nr"].string ?? ""
             let idNumber = Int(id) ?? 0
             let nr = Int(typeNR) ?? Int.max
             let dateformatter = DateFormatter()
+            let price = priceList[typeLong]
+                name = parseMensaMenu(name).name
+            let details = parseMensaMenu(name)
             dateformatter.dateFormat = "yyyy-MM-dd"
             let dateAsDate = dateformatter.date(from: date) ?? Date()
-            let newMenu = CafeteriaMenu(id: idNumber, date: dateAsDate, typeShort: typeShort, typeLong: typeLong, typeNr: nr, name: name)
+            let newMenu = CafeteriaMenu(id: idNumber, date: dateAsDate, typeShort: typeShort, typeLong: typeLong, typeNr: nr, name: name, price: price, details: details)
             mensa.addMenu(newMenu)
             CafeteriaMenuManager.cafeteriaMenus.append(newMenu)
         }
     }
     
+    func parseMensaMenu(_ name: String) -> MenuDetail {
+        
+        let pattern = "(?:(?<=\\((?:[[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?,)?)|(?<=,(?:[[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?,)?))([[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?)(?=(?:,[[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?)*\\))"
+        var notMatchedEmoji: [String] = []
+        var matchedAnnotations: [String] = []
+        var matchedEmoji: [String] = []
+        var matchedDescriptions: [String] = []
+        let output = NSMutableString(string: name)
```

Comment:
Ewww. By using NSMutableString. I get that you're using NSRange here. Can't you get it out. Using regular strings and the Swift RegularExpression class. That should give you native Swift Ranges